lua/swampshop/vgui/customizer.lua

```diff
@@ -296,523 +263,538 @@ local function Container(parent, label)
         end
     end
 
+
     LabelMaker(pane, label, true)
+    ]]
 
     return pane
 end
 
-function PANEL:SetupControls()
-    for k, v in pairs(self.controlzone:GetChildren()) do
-        v:Remove()
+function PANEL:AddContainer(section, label)
+    local parent = self.Sections[section]
+
+    if (not IsValid(parent)) then
+        parent = self:AddSection(section)
+        ErrorNoHaltWithStack("container defined before section!!!! please fix this")
     end
 
-    local wear_container = vgui.Create("DScrollPanel", self.controlzone)
-    wear_container:Dock(LEFT)
-    wear_container:SetWide(SS_GetMainGridDivision(2))
-    wear_container:DockMargin(0, 0, SS_COMMONMARGIN, 0)
-    wear_container.VBar:DockMargin(SS_COMMONMARGIN, 0, SS_COMMONMARGIN, 0)
-    SS_SetupVBar(wear_container.VBar)
-    wear_container.VBar:SetWide(SS_SCROLL_WIDTH)
-    local pone = LocalPlayer():IsPony()
-    local suffix = pone and "_p" or "_h"
-    local itmcw = self.item.configurable.wear
+    local pane = Container(parent, label)
 
-    if (self.item.configurable or {}).wear then
-        --LabelMaker(wearzone, "Position (" .. (pone and "pony" or "human") .. ")", true)
-        local pane = Container(wear_container, "Attachment")
-        local cnt = vgui.Create("DPanel", pane)
-        cnt:Dock(TOP)
-        cnt:SetTall(24)
-        cnt.Paint = noop
-        ATTACHSELECT = vgui.Create("DComboBox", cnt)
-        ATTACHSELECT:SetValue((self.item.cfg[self.wear] or {}).attach or (pone and (self.item.wear.pony or {}).attach) or self.item.wear.attach)
-
-        for k, v in pairs(SS_Attachments) do
-            ATTACHSELECT:AddChoice(k)
-        end
+    return pane
+end
 
-        ATTACHSELECT:SetTall(32)
-        ATTACHSELECT:Dock(FILL)
-        ATTACHSELECT.Paint = SS_PaintBG
+function PANEL:AddColorOption(section, keys, label)
+    local item = self.item
+    local suffix = self.WEARSUFFIX
+    local options = GetNestedInfo(item, keys)
+    local pane = self:AddContainer(section, label)
+    local cv = GetNestedProperty(item, keys,TYPE_VECTOR)
+
+    if (not isvector(cv)) then
+        cv = Vector(1, 1, 1)
+    end
+
+    local cvm = math.max(1, cv.x, cv.y, cv.z)
+    local PSCMixer = vgui.Create("DColorMixer", pane)
```

Comment:
Can you use vgui()

Reply:
i can't specify the parent if it's done with an existing stack, and i don't think its position in the stack would quite predictable enough for me to just parent it to whatever is above in the stack. vgui( seems a lot more useful for static layout.